src/jsonLanguageService.ts

```diff
@@ -50,19 +50,49 @@ export interface LanguageService {
 	getColorPresentations(document: TextDocument, doc: JSONDocument, color: Color, range: Range): ColorPresentation[];
 	doHover(document: TextDocument, position: Position, doc: JSONDocument): Thenable<Hover | null>;
 	format(document: TextDocument, range: Range, options: FormattingOptions): TextEdit[];
+	sort(document: TextDocument, options: FormattingOptions): string;
 	getFoldingRanges(document: TextDocument, context?: FoldingRangesContext): FoldingRange[];
 	getSelectionRanges(document: TextDocument, positions: Position[], doc: JSONDocument): SelectionRange[];
 	findDefinition(document: TextDocument, position: Position, doc: JSONDocument): Thenable<DefinitionLink[]>;
 	findLinks(document: TextDocument, doc: JSONDocument): Thenable<DocumentLink[]>;
 }
 
+function sortJSON(jsonObject: Record<any, any>): Record<any, any> {
+	let sortedJSON : Record<any, any> = {};
+	if (jsonObject.constructor.name === "Array") {
+		sortedJSON = jsonObject;
+		sortedJSON.forEach(function (value : any, index : number) {
+			sortedJSON[index] = sortJSON(value);
+		});
+	} else if (jsonObject.constructor.name === "Object") {
```

Comment:
we avoid using the constructor name use `val === 'object' && val !== null && !Array.isArray(val)` for which we already have a utility function in `objects.ts`: `isObject`